hyperspy/io_plugins/zspy.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2021 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import warnings
+import logging
+from functools import partial
+from collections import MutableMapping
+
+import zarr
+from zarr import Array, Group
+import numpy as np
+import dask.array as da
+from hyperspy.io_plugins.hspy import version
+import numcodecs
+
+
+from hyperspy.io_plugins.hierarchical import HierarchicalWriter, HierarchicalReader, _overwrite_dataset
+
+_logger = logging.getLogger(__name__)
+
+
+# Plugin characteristics
+# ----------------------
+format_name = 'ZSpy'
+description = \
+    'A default file format for HyperSpy based on the zarr standard'
+full_support = False
+# Recognised file extension
+file_extensions = ['zspy']
+default_extension = 0
+# Writing capabilities
+non_uniform_axis = True
+writes = True
+
```

Comment:
Should it have a reader version too here, which match the `hspy` one?

Reply:
Yea I just import it from the hspy file in line 28.